### DISCUSSION TITLE
Doubt in ERC20Mock

### DISCUSSION THREAD

**Discussion by Not-Sarthak** [Cyfrin/foundry-full-course-cu#2318] [Q&A] [answered]
How and Why are 4 arguments being passed in the function below taken from one of the files of StableCoin repo:

```
 function getOrCreateAnvilEthConfig() public returns (NetworkConfig memory anvilNetworkConfig) {
        // Check to see if we set an active network config
        if (activeNetworkConfig.wethUsdPriceFeed != address(0)) {
            return activeNetworkConfig;
        }

        vm.startBroadcast();
        MockV3Aggregator ethUsdPriceFeed = new MockV3Aggregator(DECIMALS, ETH_USD_PRICE);
        ERC20Mock wethMock = new ERC20Mock("WETH", "WETH", msg.sender, 1000e8);

        MockV3Aggregator btcUsdPriceFeed = new MockV3Aggregator(DECIMALS, BTC_USD_PRICE);
        ERC20Mock wbtcMock = new ERC20Mock("WBTC", "WBTC", msg.sender, 1000e8);
        vm.stopBroadcast();

        anvilNetworkConfig = NetworkConfig({
            wethUsdPriceFeed: address(ethUsdPriceFeed), // ETH / USD
            weth: address(wethMock),
            wbtcUsdPriceFeed: address(btcUsdPriceFeed),
            wbtc: address(wbtcMock),
            deployerKey: DEFAULT_ANVIL_PRIVATE_KEY
        });
    }
```
    
On navigating to ERC20Mock it expects zero arguments and hence returns me an error as:

```
    Wrong argument count for function call: 4 arguments given but expected 0.solidity(6160)
```

Reference:
https://github.com/Cyfrin/foundry-defi-stablecoin-cu/blob/main/script/HelperConfig.s.sol

**Comment by EngrPips** [comment]
Hello @Not-Sarthak, All these packages we use get updated occasionally, and stuff changes, which is not a big deal. Whenever you notice a difference between what is being taught in the course and the codebase you are working with, please always visit https://github.com/Cyfrin/foundry-full-course-cu/blob/main/chronological-updates.md first to confirm If the changes have been accounted for.

**Reply by Not-Sarthak** [reply]
yep I did check there. I thought Discussion is the quickest way to get an answer and hence I directly bump in mostly. 
Apologies if this isn't the most correct way! 🫡

**Reply by EngrPips** [reply]
I mean, if you had already checked there and seen the answer but still want to discuss it, no worries. I am happy to discuss with you, Friend.

**Comment by EngrPips** [comment, marked as answer]
This issue you raised has already addressed in https://github.com/Cyfrin/foundry-full-course-cu/blob/main/chronological-updates.md as shown in the image below.

<img width="1440" alt="Screenshot 2024-08-25 at 20 40 54" src="https://github.com/user-attachments/assets/45743e64-1bc1-4a76-9e53-cbc26a5162de">

**Comment by Totiruzi** [comment]
@EngrPips I read the comments and instructions though I am working with the latest version of Openzeppelin . I copied past the ERC20Mock file into my project and used that. 
Would like to ask, if you want to work with the updated version, how would you about using that. 
Thanks )

**Reply by EngrPips** [reply]
Working with an updated version is as easy as working with the previous version. You just need to understand the updated version and work with according to how it is designed.

**Reply by Akashjayan1999** [reply]
if you using the latest version of openzepplin,
you can update it as
 ```solidity
   vm.startBroadcast();
        MockV3Aggregator ethUsdPriceFeed = new MockV3Aggregator(DECIMALS, ETH_USD_PRICE);
        ERC20Mock wethMock = new ERC20Mock();
        wethMock.mint(msg.sender, 1000e18);



        MockV3Aggregator btcUsdPriceFeed = new MockV3Aggregator(DECIMALS, BTC_USD_PRICE);
        ERC20Mock wbtcMock = new ERC20Mock();
        wbtcMock.mint(msg.sender, 1000e8);
        vm.stopBroadcast();
       ```

**Reply by EngrPips** [reply]
Great Heads-up
